docs/proposal/2022-06-14-bindings.md

```diff
@@ -0,0 +1,681 @@
+
+#### Top level Namespace property
+
+In the proposed solution, the **namespace** property is defined separately for executors and notifiers. This approach provides fine-grained configuration. You can specify allowed namespace independently, so you can watch for events in all Namespaces but allow `kubectl` usage only in `dev` Namespace.
+
+Unfortunately it doesn't come without any cost. If you want to have a dedicated bindings for Team A, which narrows-down all `notifiers` to `exectors` to the `team-a` Namespace, you need to configure that multiple times. It may be error-prone.
+To solve that we can extract the **namespace** property to top level. In this case it will be common for all bindings:
+
+```yaml
+communications: # allows to have multiple slacks, or ES configurations
+  - name: tenant-workspace
+    slack:
+      token: 'SLACK_API_TOKEN'
+      # customized notifications
+      channels:
+        - name: "#nodes"
+          namespace:
+            include: ["team-a"]
+          bindings:
+            notifiers:
+              - "nodes-errors"
+            executors:
+              - "kubectl-read-only"
+              - "helm-full-access"
+
+executors:
+  - name: kubectl-read-only
+    kubectl:
+      commands:
+        verbs: ["api-resources", "...", "auth"]
+        resources: ["deployments", "..", "nodes"]
+  - name: helm-full-access
+    helm:
+      commands:
+        verbs: ["list", "delete", "install"]
+
+notifiers:
+  - name: nodes-errors
+   kubernetes:
+     resources:
+       - name: v1/nodes
+         events:
+           - error
+```
+
+</details>
+
+## Consequences
+
+This section described necessary changes if proposal will be accepted.
+
+### Minimum changes
+
+1. The `resources` notifications are moved under `notifiers[].kubernetes[].resources`.
+2. Kubectl executor moved under `executors[].kubectl`.
+3. The `namespaces.include` and `namespaces.exclude` properties are added to the `kubectl` executor.
+4. The `namespaces.include` and `namespaces.exclude` properties are added to `notifiers[].kubernetes[]`.
+5. The `resource_config.yaml` and `comm_config.yaml` are merged into one, but you can provide config multiple times. In the same way, as Helm takes the `values.yaml` file. It's up to the user how it will be split.
+6. Update documentation about configuration.
+
+### Follow-up changes
+
+1. Change selector for all Namespaces from `all` to `@all`.
+2. Add full channel/user indicator - `@` or `#`.
+3. Recommendations are merged under notifications.
+4. Filters are removed and existing one are moved under `notifiers[].{name}`.
```

Comment:
So are they removed or not? Sorry, I don't understand this point - cause right now there's no configuration for each filter, so what is moved? 🤔

Reply:
I wanted to remove this functionality completely and the existing ones renamed as notifiers with configuration added under `notifiers[].{name}`.

However, we don't have the plugin system in place, so I rephrased it and made optional. Maybe it will be better to do that in the scope of the plugin system.

----
5. **Optional**: [Filters](https://www.botkube.io/filters/) are renamed to `notifiers` and configuration is added under `notifiers[].{name}`.
    > **NOTE:** In the future, [Filters](https://www.botkube.io/filters/) should be completely removed from the BotKube and replaced with the plugin system.
----